Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -1006,13 +1006,14 @@ - (void)_updateScrollViewForTransaction:(const WebKit::RemoteLayerTreeTransactio
         LOG_WITH_STREAM(VisibleRects, stream << " updating scroll view with pageScaleFactor " << layerTreeTransaction.pageScaleFactor());
 
         // When web-process-originated scale changes occur, pin the
-        // vertical scroll position to the top edge of the content,
+        // scroll position to the top edge of the content,
         // instead of the center (which UIScrollView does by default).
         CGFloat contentOffsetY = [_scrollView contentOffset].y * layerTreeTransaction.pageScaleFactor() / [_scrollView zoomScale];
+        CGFloat contentOffsetX = [_scrollView contentOffset].x * layerTreeTransaction.pageScaleFactor() / [_scrollView zoomScale];
```

Comment:
Nit - maybe we can pull `layerTreeTransaction.pageScaleFactor() / [_scrollView zoomScale]` out into a separate local variable?

Reply:
Good call, done!